lib/fluent/plugin/out_google_cloud.rb

```diff
@@ -600,19 +605,20 @@ def write(chunk)
           entry_common_labels = \
             group_common_labels.merge(extracted_common_labels)
 
-          if entry_resource.type == CONTAINER_CONSTANTS[:resource_type]
+          is_json = false
+          if @detect_json
             # Save the timestamp if available, then clear it out to allow for
             # determining whether we should parse the log or message field.
-            timestamp = record.key?('time') ? record['time'] : nil
-            record.delete('time')
+            timestamp = record.delete('time')
             # If the log is json, we want to export it as a structured log
             # unless there is additional metadata that would be lost.
-            is_json = false
-            if record.length == 1 && record.key?('log')
-              record_json = parse_json_or_nil(record['log'])
-            end
-            if record.length == 1 && record.key?('message')
-              record_json = parse_json_or_nil(record['message'])
+            record_json = nil
+            if record.length == 1
+              %w(log message msg).each do |field|
```

Comment:
what's the difference between `msg` and `message`. Trying to understand why we need both.

Reply:
They are just two of the three different field names that may contain the log text. Because different systems use different names by default, we should be checking all three.